haystack/preview/components/generators/openai/gpt.py

```diff
@@ -84,12 +86,23 @@ def __init__(
             - `logit_bias`: Add a logit bias to specific tokens. The keys of the dictionary are tokens and the
                 values are the bias to add to that token.
         """
-        self.api_key = api_key
+        if api_key is None:
+            try:
+                api_key = os.environ["OPENAI_API_KEY"]
+            except KeyError as e:
+                raise ValueError(
+                    "GPTGenerator expects an OpenAI API key. "
+                    "Set the OPENAI_API_KEY environment variable (recommended) or pass it explicitly."
+                ) from e
+        openai.api_key = api_key
```

Comment:
With the current approach, it would be ignored...

What's the best approach, according to you?

With the [upcoming OpenAI SDK](https://github.com/openai/openai-python/discussions/631), things will become clearer (they allow creating a client instance).

Reply:
I would check if `openai.api_key` is set too before actually failing.

Reply:
makes sense...

Reply:
The question is totally valid, but let's not forget that in the [SDK upgrade](https://github.com/openai/openai-python/discussions/631) they plan to remove this way of setting the key, so I would not over-engineer a solution here

Reply:
The new version is not out and we won't be moving to that as soon as it comes out, so handling this case makes sense as it enhances the UX. Also it's not overengineering, it's an extra check in the if.